python/cuml/fil/README.md

```diff
@@ -0,0 +1,74 @@
+# FIL - RAPIDS Forest Inference Library
+
+The Forest Inference Library provides a lightweight, flexible API to
+infer (predict) results from a tree-based model ensemble on GPU. The
+tree ensemble can be either gradient-boosted decision tree (GBDT) or
+random forest (RF) models trained in XGBoost, cuML, scikit-learn, or
+LightGBM.
+
+# Code sample
+
+Starting with an XGBoost classification model saved in the file
+"xgb.mod," we want to use that model to infer on a large dataset of
+test samples.
+
+```python
+from cuml import ForestInference
+
+fm = ForestInference.load(filename=model_path,
+                          output_class=True,
+                          threshold=0.50,
+                          model_type='xgboost')
+
+X = ... load test samples as a numpy or cupy array ...
+
+y_out = fm.predict(X)
+
+```
+
+See [the sample notebook](https://github.com/rapidsai/cuml/blob/branch-0.16/notebooks/forest_inference_demo.ipynb) for much more detail and runnable samples.
+
+# Features
+
+* Input model source: XGBoost (binary format), cuML RandomForest, scikit-learn RandomForest, LightGBM
+* Model types: Regression, Binary Classification, Multi-class Classification (for Random Forests, but not GBDTs)
+* Tree storage types: Dense or sparse tree storage (see Sparse Forests with FIL blog below)
+* Input formats: Dense, row-major, FP32 arrays on GPU or CPU (e.g. NumPy, cuPy, or other data formats supported by cuML)
+* High performance batch inference
+* Input parsing based on (Treelite)[https://github.com/dmlc/treelite]
+
+Upcoming features:
+
+* Support for multi-class GBDTs is planned for RAPIDS 0.16
+
+# Benchmarks and performance notes
+
+(1) The core data format supported by FIL is an FP32, row-major array on
+GPU. All other input types will be automatically converted to this
+format internally, but you will get the lowest latency if you use that
+format to start with.
+
+(2) FIL is optimized for high-throughput, batch inference, so its
+performance benefits become more pronounced as the size of the test
+data X grows. Larger, more complex models (e.g. those with more trees)
+will also see a greater boost as they can fully occupy a large GPU.
+
+The chart below shows how performance (measured in microseconds per
+row) varies as the number of input rows increases, comparing both
+CPU-based inference (XGBoost CPU inference, and the optimized treelite
+library) and GPU-based inference (XGBoost and FIL).
+
+![FIL Performance Chart](./fil_performance_nrows.png)
```

Comment:
could you share the gnuplot script? I would like to propose a couple of quick changes to make it a bit easier to read

Reply:
This is from the Medium blog post: https://medium.com/rapids-ai/rapids-forest-inference-library-prediction-at-100-million-rows-per-second-19558890bc35. I think Andy created the graph?

Reply:
I'd like to propose we go ahead with the current chart and then improve it in a future rev, as we have previously shared this chart in the blog as well.

Reply:
yes, makes sense.